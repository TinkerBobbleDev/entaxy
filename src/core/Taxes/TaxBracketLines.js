import React from 'react'
import { Line } from '@vx/shape';
import { Point } from '@vx/point';
import { taxBracketData } from './TaxBrackets'
import red from 'material-ui/colors/red';
import blueGrey from 'material-ui/colors/blueGrey';

const TaxBracketLines = (
  year,
  province,
  xScale,
  yScale,
  margin,
  width,
  height
) => {
  const rows = []
  const yMax = height - margin.top - margin.bottom
  const xMax = width - margin.left - margin.right
  const data = taxBracketData(year, province)

  for(var d=0; d < data.length; d++){
    var left = xScale(data[d].income)
    var right = data[d+1] ? xScale(data[d+1].income) : xMax

    rows.push(
      <Line
        key={"bracket-" + d}
        from={new Point({x: left, y: yMax})}
        to={new Point({x: left, y: 10})}
        stroke={red[100]}
        strokeWidth={1}
      />
    )
    rows.push(
      <text
        fill={blueGrey[800]}
        textAnchor="middle"
        x={left + (right - left)/2}
        y={0}
        dy=".33em"
        fontSize={9}
        fontFamily={'Roboto'}
      >
      {data[d].tax}%
    </text>

    )
  }
  return  <g key={"TaxBracketLines"}>{rows}</g>
}

export default TaxBracketLines
