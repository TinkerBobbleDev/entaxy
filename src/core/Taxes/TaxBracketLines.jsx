import React from 'react'
import PropTypes from 'prop-types'
import { Line } from '@vx/shape'
import { Point } from '@vx/point'
import red from 'material-ui/colors/red'
import blueGrey from 'material-ui/colors/blueGrey'
import { TaxBrackets, taxBracketData } from './lib/TaxBrackets'

const TaxBracketLines = ({
  country,
  year,
  region,
  xScale,
  margin,
  width,
  height
}) => {
  const rows = []
  const yMax = height - margin.top - margin.bottom
  const xMax = width - margin.left - margin.right
  const data = taxBracketData(TaxBrackets[country][year], region)

  for (let d = 0; d < data.length; d += 1) {
    const left = xScale(data[d].income)
    const right = data[d + 1] ? xScale(data[d + 1].income) : xMax

    const line = (
      <Line
        key={`bracket-line${d}`}
        from={new Point({ x: left, y: yMax })}
        to={new Point({ x: left, y: 10 })}
        stroke={red[100]}
        strokeWidth={1}
      />
    )

    const text = (
      <text
        key={`bracket-text${d}`}
        fill={blueGrey[800]}
        textAnchor="middle"
        x={left + ((right - left) / 2)}
        y={0}
        dy=".33em"
        fontSize={9}
        fontFamily="Roboto"
      >
        {data[d].tax}
      </text>
    )

    rows.push(line)
    rows.push(text)
  }
  return <g key="TaxBracketLines">{rows}</g>
}

TaxBracketLines.propTypes = {
  country: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
  region: PropTypes.string,
  xScale: PropTypes.func.isRequired,
  margin: PropTypes.object.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired
}

TaxBracketLines.defaultProps = {
  region: null
}

export default TaxBracketLines
