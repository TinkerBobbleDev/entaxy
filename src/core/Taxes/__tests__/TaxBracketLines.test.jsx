import React from 'react'
import renderer from 'react-test-renderer'
import TaxBracketLines from '../TaxBracketLines'

describe('TaxBracketLines', () => {
  it('matches snapshot', () => {
    const margin = {
      top: 20,
      bottom: 40,
      left: 70,
      right: 0
    }
    const xScaleMoch = jest.fn()
    const component = renderer.create((
      <TaxBracketLines
        income={1000}
        country="Canada"
        year={2017}
        region="Ontario"
        xScale={xScaleMoch}
        margin={margin}
        width={100}
        height={100}
        classes={{ rrsp: {} }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
