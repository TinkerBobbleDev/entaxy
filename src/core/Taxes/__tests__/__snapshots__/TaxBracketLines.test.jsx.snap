// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaxBracketLines matches snapshot 1`] = `
<g>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    5.05
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    20.05
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    24.15
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    29.65
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    31.66
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    37.16
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    40.16
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    41.16
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    45.16
  </text>
  <line
    className="vx-line"
    stroke="#ffcdd2"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={10}
  />
  <text
    dy=".33em"
    fill="#37474f"
    fontFamily="Roboto"
    fontSize={9}
    textAnchor="middle"
    x={NaN}
    y={0}
  >
    46.16
  </text>
</g>
`;
